tock/api/renderers.py

```diff
@@ -16,3 +21,38 @@ def render(self, data, media_type=None, renderer_context=None):
         if not self.headers and len(data) > 0:
             self.headers = data[0].keys()
         return super(PaginatedCSVRenderer, self).render(data, media_type, renderer_context)
+
+def stream_csv(queryset, serializer):
+    """
+    Stream data as CSV, given an interable queryset and a DRF
+    Serializer instance with a .fields dict and .to_representation()
+    method.
+    """
+    rows = map(serializer.to_representation, queryset)
```

Comment:
Django tends to cache results from querysets, which might blow out your system memory for huge result sets. One way to avoid that is to use `queryset.iterator()`. Kinda related - `DEBUG == True` makes django log a lot more about queries; if you ever run into trouble with memory during development, this might be why.